tool/grind.dart

```diff
@@ -44,6 +44,9 @@ void main(List<String> args) {
   pkg.standaloneName.value = "dart-sass";
   pkg.githubUser.fn = () => Platform.environment["GH_USER"];
   pkg.githubPassword.fn = () => Platform.environment["GH_TOKEN"];
+  // Limit the methods exposed as ESM exports to those that don't require
+  // filesystem access
+  pkg.jsEsmExports.value = {'compileString', 'compileStringAsync'};
```

Comment:
@nex3 with this configuration, `grinder pkg-npm-dev` produces this code in `sass.default.js`:

```javascript
import * as util from "util"
import * as immutable from "immutable"
import * as fs from "fs"
import * as stream from "stream"
import * as _cliPkgModule from "./sass.dart.js";

// Bundlers may try to resolve this file eagerly, in which case
// sass.dart.js won't be able to accurately detect that it's being imported
// rather than required. Fortunately, in that case, exports will be defiend and
// we can load the library that way instead.
const _cliPkgLibrary =
    'load' in _cliPkgModule ? _cliPkgModule : window._cliPkgExports;
delete window._cliPkgExports;
const _cliPkgExports = {};
_cliPkgLibrary.load({util, immutable, fs, stream}, _cliPkgExports);

export const compileString = _cliPkgExports.compileString;
export const compileStringAsync = _cliPkgExports.compileStringAsync;
```

That _almost_ works. I just had to change from `import * as _cliPkgModule` to `import _cliPkgModule`. Otherwise accessing the `load` function is not possible.

Reply:
On Slack, Natalie explained this might be related to the bundler I'm using to test. I tried without bundlers, and this works as expected.